lib/libzpool/kernel.c

```diff
@@ -818,9 +818,8 @@ random_get_pseudo_bytes(uint8_t *ptr, size_t len)
 int
 ddi_strtoul(const char *hw_serial, char **nptr, int base, unsigned long *result)
 {
-	char *end;
-
-	*result = strtoul(hw_serial, &end, base);
+	char* end;
+	*result = strtoul(hw_serial, nptr, base);
```

Comment:
Pleasing you found this as well, sorry I didn't see it before fixing it. Other ZFS code can pass NULL as result, and Win32 will abort on it, so it needed slightly different fix,